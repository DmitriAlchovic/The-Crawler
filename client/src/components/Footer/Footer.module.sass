.footer-body
    display: flex
    justify-content: space-between
    margin: 29px 101px 36px 133px
    border-top: 2px solid rgba(0, 0, 0, 0.05) 
    padding-top: 33px 
    font-size: 16px

    .logo
        color: #3bb77e
        font-weight: 700
        font-size: 28px
        margin-bottom: 34px

    .contacts
        margin-bottom: 29px

    .icon
        margin-right: 3px

    .info-header
        font-size: 24px
        font-weight: 600px
        margin-bottom: 43px

    .info
        margin-bottom: 15px
