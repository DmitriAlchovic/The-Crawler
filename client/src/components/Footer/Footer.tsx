import React from 'react';
import { Link } from 'react-router-dom';
import payment from '../../assets/payment.svg';
import facebook from '../../assets/icon _facebook_.svg';
import linkedin from '../../assets/icon _linkedin_.svg';
import telegram from '../../assets/icon _telegram plane_.svg';
import insta from '../../assets/icon _instagram_.svg';
import twitter from '../../assets/icon _twitter_.svg';
import location from '../../assets/icon_location.svg';
import phone from '../../assets/icon_phone.svg';
import envelope from '../../assets/icon_envelope.svg';
import clock from '../../assets/icon_clock.svg';
import logo from '../../assets/logo.svg';
import styles from './Footer.module.sass';
import paths from '../../utils/paths';

export default function Footer() {
  return (
    <div>
      <div className={styles['footer-body']}>
        <div>
          <div className={styles.logo}>
            <img className={styles.icon} alt="logo" src={logo} />
            Grocery
          </div>
          <div className={styles.contacts}>
            <img className={styles.icon} alt="location" src={location} />
            Address: 4242 Road House
          </div>
          <div className={styles.contacts}>
            <img className={styles.icon} alt="phone" src={phone} />
            Call Us: 1234-5678
          </div>
          <div className={styles.contacts}>
            <img className={styles.icon} alt="mail" src={envelope} />
            Email: grocery@contact.com
          </div>
          <div>
            <img className={styles.icon} alt="clock" src={clock} />
            Work hours: 8:00 - 20:00, Sunday - Thursday
          </div>
        </div>
        <div>
          <div className={styles['info-header']}>Useful links</div>
          <Link to={`${paths.categoryPage}hot/1`}><div className={styles.info}>Hot deals</div></Link>
          <Link to={`${paths.categoryPage}promotions/1`}><div className={styles.info}>Promotions</div></Link>
          <Link to={`${paths.categoryPage}new/1`}><div className={styles.info}>New products</div></Link>
        </div>
      </div>
      <div className={styles['footer-body']}>
        <div>Â© 2022, All rights reserved</div>
        <div>
          <img alt="payment systems" src={payment} />
        </div>
        <div>
          <img alt="facebook" src={facebook} />
          <img alt="linkedin" src={linkedin} />
          <img alt="telegram" src={telegram} />
          <img alt="instagram" src={insta} />
          <img alt="twitter" src={twitter} />
        </div>
      </div>
    </div>
  );
}
