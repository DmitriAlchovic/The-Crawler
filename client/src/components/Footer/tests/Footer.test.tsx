import React from 'react';
import { render } from '@testing-library/react';
import * as reduxHooks from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Footer from '../Footer';

jest.mock('react-redux');

describe('Info', () => {
  it('should create empty component', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue('');
    const component = render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
