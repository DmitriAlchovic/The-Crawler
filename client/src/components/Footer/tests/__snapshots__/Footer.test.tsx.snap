// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should create empty component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="footer-body"
        >
          <div>
            <div
              class="logo"
            >
              <img
                alt="logo"
                class="icon"
                src="logo.svg"
              />
              Grocery
            </div>
            <div
              class="contacts"
            >
              <img
                alt="location"
                class="icon"
                src="icon_location.svg"
              />
              Address: 4242 Road House
            </div>
            <div
              class="contacts"
            >
              <img
                alt="phone"
                class="icon"
                src="icon_phone.svg"
              />
              Call Us: 1234-5678
            </div>
            <div
              class="contacts"
            >
              <img
                alt="mail"
                class="icon"
                src="icon_envelope.svg"
              />
              Email: grocery@contact.com
            </div>
            <div>
              <img
                alt="clock"
                class="icon"
                src="icon_clock.svg"
              />
              Work hours: 8:00 - 20:00, Sunday - Thursday
            </div>
          </div>
          <div>
            <div
              class="info-header"
            >
              Useful links
            </div>
            <a
              href="/category/hot/1"
            >
              <div
                class="info"
              >
                Hot deals
              </div>
            </a>
            <a
              href="/category/promotions/1"
            >
              <div
                class="info"
              >
                Promotions
              </div>
            </a>
            <a
              href="/category/new/1"
            >
              <div
                class="info"
              >
                New products
              </div>
            </a>
          </div>
        </div>
        <div
          class="footer-body"
        >
          <div>
            © 2022, All rights reserved
          </div>
          <div>
            <img
              alt="payment systems"
              src="payment.svg"
            />
          </div>
          <div>
            <img
              alt="facebook"
              src="icon _facebook_.svg"
            />
            <img
              alt="linkedin"
              src="icon _linkedin_.svg"
            />
            <img
              alt="telegram"
              src="icon _telegram plane_.svg"
            />
            <img
              alt="instagram"
              src="icon _instagram_.svg"
            />
            <img
              alt="twitter"
              src="icon _twitter_.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="footer-body"
      >
        <div>
          <div
            class="logo"
          >
            <img
              alt="logo"
              class="icon"
              src="logo.svg"
            />
            Grocery
          </div>
          <div
            class="contacts"
          >
            <img
              alt="location"
              class="icon"
              src="icon_location.svg"
            />
            Address: 4242 Road House
          </div>
          <div
            class="contacts"
          >
            <img
              alt="phone"
              class="icon"
              src="icon_phone.svg"
            />
            Call Us: 1234-5678
          </div>
          <div
            class="contacts"
          >
            <img
              alt="mail"
              class="icon"
              src="icon_envelope.svg"
            />
            Email: grocery@contact.com
          </div>
          <div>
            <img
              alt="clock"
              class="icon"
              src="icon_clock.svg"
            />
            Work hours: 8:00 - 20:00, Sunday - Thursday
          </div>
        </div>
        <div>
          <div
            class="info-header"
          >
            Useful links
          </div>
          <a
            href="/category/hot/1"
          >
            <div
              class="info"
            >
              Hot deals
            </div>
          </a>
          <a
            href="/category/promotions/1"
          >
            <div
              class="info"
            >
              Promotions
            </div>
          </a>
          <a
            href="/category/new/1"
          >
            <div
              class="info"
            >
              New products
            </div>
          </a>
        </div>
      </div>
      <div
        class="footer-body"
      >
        <div>
          © 2022, All rights reserved
        </div>
        <div>
          <img
            alt="payment systems"
            src="payment.svg"
          />
        </div>
        <div>
          <img
            alt="facebook"
            src="icon _facebook_.svg"
          />
          <img
            alt="linkedin"
            src="icon _linkedin_.svg"
          />
          <img
            alt="telegram"
            src="icon _telegram plane_.svg"
          />
          <img
            alt="instagram"
            src="icon _instagram_.svg"
          />
          <img
            alt="twitter"
            src="icon _twitter_.svg"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
